src/base_output_updater.cpp

```diff
@@ -36,6 +36,57 @@ void compute_delta_z_output(std::vector<float> &mu_a, std::vector<float> &var_a,
     }
 }
 
+void compute_delta_z_noise(std::vector<float> &mu_a, std::vector<float> &var_a,
+                           std::vector<float> &jcb, std::vector<float> &obs,
+                           int start_chunk, int end_chunk,
+                           std::vector<float> &delta_mu,
+                           std::vector<float> &delta_var) {
+    const float zero_pad = 0.0f;
+
+    for (int col = start_chunk; col < end_chunk; col += 2) {
+        float mu_V2_bar_tilde = mu_a[col + 1];
```

Comment:
I would take out all the outpout at the beginning here at `col` and `col + 1` positions and adding comments what are those variables

Reply:
I explained the with comments most of the things, but I believe that taking out the output and other variables would make the function even more unreadable.